server/src/main/scala/cromwell/CromwellEntryPoint.scala

```diff
@@ -78,13 +79,24 @@ object CromwellEntryPoint extends GracefulStopSupport {
     val cromwellClient = new CromwellClient(args.host, "v2")
 
     val singleSubmission = validateSubmitArguments(args)
-    val submissionFuture = () => cromwellClient.submit(singleSubmission).andThen({
-      case Success(submitted) =>
-        Log.info(s"Workflow ${submitted.id} submitted to ${args.host}")
-      case Failure(t) =>
-        Log.error(s"Failed to submit workflow to cromwell server at ${args.host}")
-        Log.error(t.getMessage)
-    })
+    val submissionFuture = () => {
+      val io = cromwellClient.submit(singleSubmission).value.attempt flatMap {
+        case Right(Left(httpResponse)) =>
+          IO.fromFuture(IO {
+            Unmarshal(httpResponse.entity).to[String] map { stringEntity =>
+              Log.info(s"Workflow failed to submit to ${args.host}: $stringEntity")
+            }
+          })
+        case Right(Right(submitted)) =>
+          Log.info(s"Workflow ${submitted.id} submitted to ${args.host}")
+          IO.unit
+        case Left(t) =>
+          Log.error(s"Failed to submit workflow to cromwell server at ${args.host}")
```

Comment:
Would it make sense to make the failure messages in the two cases more alike?

Reply:
Done